mobile/lib/services/rest_api.dart

```diff
@@ -283,6 +284,79 @@ class AirqoApiClient {
     return airQualityReadings.removeInvalidData();
   }
 
+  Future<List<FavouritePlace>> fetchFavoritePlaces(String userId) async {
+    final favoritePlaces = <FavouritePlace>[];
+    final queryParams = <String, String>{}
+      ..putIfAbsent('tenant', () => 'airqo');
+
+    try {
+      final body = await _performGetRequest(
+        queryParams,
+        "${AirQoUrls.favourites}/$userId",
+        apiService: ApiService.auth,
+      );
+
+      for (final favorite in body['favorites'] as List<dynamic>) {
+        try {
+          favoritePlaces.add(
+            FavouritePlace.fromJson(
+              favorite as Map<String, dynamic>,
+            ),
+          );
+        } catch (exception, stackTrace) {
+          await logException(
+            exception,
+            stackTrace,
+          );
+        }
+      }
+    } catch (exception, stackTrace) {
+      await logException(
+        exception,
+        stackTrace,
+      );
+    }
+
+    return favoritePlaces;
+  }
+
+  Future<bool> syncFavouritePlaces(List<FavouritePlace> favorites) async {
+    final userId = CustomAuth.getUserId();
+
+    if (userId.isEmpty) {
+      return false;
+    }
+    try {
+      Map<String, String> headers = Map.from(postHeaders);
+      headers["service"] = ApiService.auth.serviceName;
+
+      dynamic body = favorites.map((e) => e.toJson()).toList();
+      body = body.map((e) => e..remove("id")).toList();
+      body.forEach((item) {
+        item['firebase_user_id'] = userId;
+      });
```

Comment:
I thinks create a `to api json` to handle this logic. @BenjaminSsempala

Reply:
I've shifted this.
Thhanks @NoahNsimbe